### PR TITLE
rgw: rgw file use quota info as fssize

### PR DESCRIPTION
**Pull request by guihecheng**
With this one and the related commit from nfs-ganesha:
         FSAL_RGW: support rgw quota-related fs info
the df command will be able to show info of a mounted RGW-NFS(of course only with bucket quota enabled).

### CODE REVIEW

File: src/rgw/rgw_file.h [hunk 1]
```diff
@@ -1992,7 +1992,7 @@ class RGWStatBucketRequest : public RGWLibRequest,
     return bucket.creation_time;
   }
 
-  bool only_bucket() override { return false; }
```
Comment (mattbenjamin) [inline src/rgw/rgw_file.h]: probably correct, I don't mind combining this if so

**Comment by smithfarm** [comment]
Jenkins re-test this please

**Comment by guihecheng** [comment]
@mattbenjamin yes, I think it is nice to handle the >1 visible buckets fs case, maybe using the user scope quota and user scope stats(will check whether there is sth like this) ?

File: src/rgw/rgw_file.h (hunk 1 above)
Comment (guihecheng) [inline src/rgw/rgw_file.h]: IMO, we'd better keep it in a separate commit if you suspect, then if we find it harmful in the future, we could easily bisect and revert it.

**Comment by guihecheng** [comment]
@mattbenjamin updated to handle >1 bucket case, thanks.

**Comment by mattbenjamin** [comment]
@guihecheng I think this is ready--are you still happy with this?  rebase? :)

**Comment by guihecheng** [comment]
@mattbenjamin oh, sure, rebasing :)

**Comment by guihecheng** [comment]
@mattbenjamin rebased, thanks.

**Comment by guihecheng** [comment]
@mattbenjamin rebased again, thanks.

**Comment by guihecheng** [comment]
@mattbenjamin any ideas or comments ?

**Comment by mattbenjamin** [comment]
@guihecheng sorry for the delay, can you rebase?

**Comment by guihecheng** [comment]
@mattbenjamin yes, sure, I shall rebase in my spare time!

**Pull request closed without merging**